### PR TITLE
Add support for controller-api to v1.8.3

### PR DESCRIPTION
**Pull request by krunalhinguu**
### Updates

- Update capi to v1.8.3 to support kubernetes v1.31

### Related

- https://github.com/rancher/rancher/issues/46197

### CODE REVIEW

File: charts-source/capi/Chart.yaml [hunk 1]
```diff
@@ -4,19 +4,19 @@ description: capi-controller-manager compatible with Rancher Provisioning
...
   catalog.cattle.io/display-name: Rancher Provisioning CAPI Controller Manager
-  catalog.cattle.io/kube-version: '>= 1.27.0-0 < 1.31.0-0'
+  catalog.cattle.io/kube-version: '>= 1.27.0-0 < 1.32.0-0'
   catalog.cattle.io/namespace: cattle-provisioning-capi-system
   catalog.cattle.io/release-name: rancher-provisioning-capi
   catalog.cattle.io/permits-os: linux,windows
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: apps.deployment/v1
-  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.10.0-0'
+  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.11.0-0'
```
Comment (snasovich) [inline charts-source/capi/Chart.yaml:19]: Any reason we want to keep 2.9.x in the range? (Note this condition doesn't really matter much as it's only used by UI for user-deployed charts but it still makes sense to keep things clean here).

```suggestion
  catalog.cattle.io/rancher-version: '>= 2.10.0-0 < 2.11.0-0'
```

File: charts/capi/Chart.yaml [hunk 2]
```diff
@@ -4,19 +4,19 @@ description: capi-controller-manager compatible with Rancher Provisioning
...
   catalog.cattle.io/display-name: Rancher Provisioning CAPI Controller Manager
-  catalog.cattle.io/kube-version: '>= 1.27.0-0 < 1.31.0-0'
+  catalog.cattle.io/kube-version: '>= 1.27.0-0 < 1.32.0-0'
   catalog.cattle.io/namespace: cattle-provisioning-capi-system
   catalog.cattle.io/release-name: rancher-provisioning-capi
   catalog.cattle.io/permits-os: linux,windows
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: apps.deployment/v1
-  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.10.0-0'
+  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.11.0-0'
```
Comment (snasovich) [inline charts/capi/Chart.yaml:19]:
```suggestion
  catalog.cattle.io/rancher-version: '>= 2.10.0-0 < 2.11.0-0'
```

File: charts/capi/templates/clusterrole-capi-manager-role.yaml
Comment (snasovich) [inline charts/capi/templates/clusterrole-capi-manager-role.yaml:1]: Hope these updates to roles have no adverse effect on our functionality, defer to @Oats87 / @jakefhyde to confirm.

**Review by snasovich — APPROVED** [review]
Mostly LGTM, couple minor comments.

**Review by Oats87 — APPROVED** [review]
LGTM; needs extensive QA but looking here: https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.7-to-v1.8 there was not much that changed in 1.8

**Pull request merged**